### PR TITLE
NavigationView: fix LifecycleRegistry calls

### LINKED ISSUE #3024: NavigatioView. Lifecycle missing invokes
Investigate issues with [NavigationView](https://github.com/mapbox/mapbox-navigation-android/blob/master/libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java) as a `LifecycleOwner`. 
- A few missing invokes at `onPause()` and `onStop()` functions. 
- Also uses _deprecated_ calls to lifecycle registry like `lifecycleRegistry.markState(Lifecycle.State.STARTED);`

### PR DESCRIPTION
**Pull request by RingerJK**
## Description

NavigationView, fix lifecycle calls 
Fix #3024

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Please describe the PR goals. Just the stuff needed to implement the fix / feature and a simple rationale. It could contain many check points if needed

### Implementation

Please include all the relevant things implemented and also rationale, clarifications / disclaimers etc. related to the approach used. It could be as self code companion comments

## Screenshots or Gifs

Please include all the media files to give some context about what's being implemented or fixed. It's not mandatory to upload screenshots or gifs, but for most of the cases it becomes really handy to get into the scope of the feature / bug being fixed and also it's REALLY useful for UI related PRs screenshot gif

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

**Comment by RingerJK** [comment]
Order of calls lifecycle callbacks is matter: https://developer.android.com/reference/androidx/lifecycle/Lifecycle
> Lifecycle.Event.ON_CREATE, Lifecycle.Event.ON_START, Lifecycle.Event.ON_RESUME events in this class are dispatched **after** the LifecycleOwner's related method returns. Lifecycle.Event.ON_PAUSE, Lifecycle.Event.ON_STOP, Lifecycle.Event.ON_DESTROY events in this class are dispatched **before** the LifecycleOwner's related method is called. For instance, Lifecycle.Event.ON_START will be dispatched after onStart returns, Lifecycle.Event.ON_STOP will be dispatched before onStop is called. This gives you certain guarantees on which state the owner is in.

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java [hunk 1]
```diff
@@ -190,44 +190,47 @@ public void onRestoreInstanceState(Bundle savedInstanceState) {
...
   }
 
   public void onResume() {
     mapView.onResume();
-    lifecycleRegistry.markState(Lifecycle.State.RESUMED);
+    lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_RESUME);
   }
 
   public void onPause() {
+    lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_PAUSE);
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java:208]: Noticed that we're normally (`onStart` and `onResume`) calling `lifecycleRegistry.handleLifecycleEvent` after `MapView`s lifecycle but not here 🤔 Is that on purpose? Could you clarify?

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java [hunk 2]
```diff
@@ -190,44 +190,47 @@ public void onRestoreInstanceState(Bundle savedInstanceState) {
...
     mapView.onPause();
   }
 
   public void onStop() {
+    lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_STOP);
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java:213]: Same as above

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java [hunk 3]
```diff
@@ -190,44 +190,47 @@ public void onRestoreInstanceState(Bundle savedInstanceState) {
...
+   * Called to ensure the {@link MapView} is destroyed
+   * properly.
+   * <p>
+   * In an {@link Activity} this should be in onDestroy().
+   * <p>
+   * In a {@link androidx.fragment.app.Fragment}, this should
+   * be in {@link androidx.fragment.app.Fragment#onDestroyView()}.
+   */
+  public void onDestroy() {
+    lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_DESTROY);
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java:230]: Same as above? 🤔

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java (hunk 1 above)
Comment (RingerJK) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java:208]: sure, for this I left https://github.com/mapbox/mapbox-navigation-android/pull/3027#issuecomment-633531307

**Pull request merged**